src/hotspot/cpu/x86/gc/z/z_x86_64.ad

```diff
@@ -34,66 +34,66 @@ source %{
 #include "c2_intelJccErratum_x86.hpp"
 #include "gc/z/zBarrierSetAssembler.hpp"
 
-static void z_color(MacroAssembler& _masm, const MachNode* node, Register ref) {
+static void z_color(MacroAssembler* masm, const MachNode* node, Register ref) {
```

Comment:
For files already using MacroAssembler& _masm, the only change needed is this at the top:
```
undef __
#define __ _masm.
```

Reply:
I guess that doesn't work because different files are concatenated together, causing a conflict if some files expect MacroAssembler *masm.  To reduce the number of changes, couldn't we use MacroAssembler& _masm everywhere?

Reply:
Because some places were using `if (cbuf)` I ended up opting to make the parameter also a pointer instead of a reference.

> For files already using MacroAssembler& _masm, the only change needed is this at the top:

My opinion would be to use Reference or Pointer everywhere and not mix the two - to prevent confusion. But if you folks think it's best to go that way, I'm fine with it.